app/src/main/java/com/developersbreach/composeactors/ui/screens/search/SearchUIState.kt

```diff
@@ -8,6 +8,6 @@ import com.developersbreach.composeactors.data.model.Actor
  * Models the UI state for the [SearchScreen] screen.
  */
 data class SearchUIState(
-    val actorList: List<Actor> = listOf(),
+    val actorList: ArrayList<Actor> = arrayListOf(),
```

Comment:
This is is decent scenario to separate this into a state since we have conjunction of usage of objects `Actor` and `Movie`, so we can do this.

```suggestion
sealed class SearchUIState {

    data class MovieSearch(
        val movieList: List<Movie> = listOf(),
        val isSearchingResults: Boolean = false,
    ): SearchUIState()

    data class ActorSearch(
        val actorList: List<Actor> = listOf(),
        val isSearchingResults: Boolean = false,
    ): SearchUIState()
}
```